Upgrade CI docker image to ROCm 7.1

LSchwiebert opened the pull request:
This PR makes the changes necessary to add hipcub to the ROCm image and move from ROCm 6.3.1 to ROCm 7.1.0. The main reason for moving to a new version is that ROCm 7.1.0 is needed to replace the thrust::DeviceTransform() call to a hipcub::DeviceTransform() call, which is needed to move to a gpu-resident implementation.

Note 1: In the Dockerfile, the web links on lines 24 & 25 are correct. There is no separate link for 7.1 for line 25 -- only latest. Not ideal, so mentioned here.
Note 2: Not sure if everything should be 7.1.0 instead of 7.1, for instance in the yml file.

LSchwiebert committed:
Updates to the docker image files for rocm7.1

LSchwiebert committed:
Rename rocm yaml for rocm version 7

LSchwiebert committed:
Change base tag from 7.1.0 to 7.1

sethrj commented:
Thanks Loren! I'm going to try to building this today and upgrading the CI image accordingly.

sethrj committed:
Disable incompatible packages

sethrj committed:
Update rocm image

sethrj committed:
Update with rocrand etc included in image

sethrj committed:
Update dockerfile that corresponds to 11-04 image

pre-commit-ci[bot] committed:
Apply fixes from pre-commit hooks: see detailed commit message →

This commit means that you must (please!) install pre-commit
on your development machine and run `pre-commit install --install-hooks`.
For more information, see
https://celeritas-project.github.io/celeritas/user/development/style.html#formatting

Autogenerated: https://pre-commit.ci

sethrj committed:
fixup! Update dockerfile that corresponds to 11-04 image

sethrj committed:
Merge branch 'develop' into docker-update

sethrj reviewed (approved):
There's definitely some follow-on work with upgrading the spack toolchain 😅 in a perfect world we could use something like https://github.com/eic/containers
